<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>495</width>
    <height>688</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>300</height>
      </size>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>继承QThread线程</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButton_qthread1">
         <property name="text">
          <string>QThread run</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pushButton_qthread1_getSomething">
         <property name="text">
          <string>getSomething</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButton_qthreadQuit">
         <property name="text">
          <string>QThread quit</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_qthread1_setSomething">
         <property name="text">
          <string>setSomething</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="pushButton_qthreadTerminate">
         <property name="text">
          <string>QThread Terminate</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="pushButton_doSomthing">
         <property name="text">
          <string>doSomting</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QPushButton" name="pushButton_qthreadExit">
         <property name="text">
          <string>QThread exit</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="pushButton_qthreadRunLocal">
         <property name="text">
          <string>QThread run local</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>继承QObject线程</string>
      </attribute>
      <widget class="QPushButton" name="pushButton_qobjectStart">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>30</y>
         <width>221</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>QObject-&gt;moveToThread run</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_qobjectStop">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>120</y>
         <width>191</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>stop from widget</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_qobjectStart_2">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>60</y>
         <width>221</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>QObject-&gt;moveToThread run2</string>
       </property>
      </widget>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>heartBeat</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QProgressBar" name="progressBar_heart">
       <property name="value">
        <number>24</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QProgressBar" name="progressBar">
     <property name="value">
      <number>24</number>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTextBrowser" name="textBrowser">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>300</height>
      </size>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
